geckomat/utilities/loadDatabases.m

```diff
@@ -1,29 +1,37 @@
-function [uniprotDB,keggDB] = loadDatabases(taxonID,keggID)
+function [uniprotDB,keggDB] = loadDatabases(modelAdapter,taxonID,keggID)
 
 geckoPath = findGECKOroot();
-if nargin<2
-    param=getModelParameters();
+param=modelAdapter.getParameters();
+if nargin<3
     keggID=param.keggID;
 end
-if nargin<1
+if nargin<2
     taxonID=num2str(param.taxonID); % In case it is numeric
 end
 
 %Uniprot
-if ~exist(fullfile(geckoPath,'databases','uniprotDB.tsv'))
-    disp(['Downloading Uniprot data for taxonomic ID ' taxonID '.'])
-    url = ['https://www.uniprot.org/uniprot/?query=taxonomy:' taxonID ...
-        '&format=tab&compress=no&columns=id,genes(OLN),genes(PREFERRED),ec,database(GeneID),database(RefSeq),mass,sequence'];
-    uniprotDL = webread(url);
-    fid=fopen(fullfile(geckoPath,'databases','uniprotDB.tsv'),'w');
-    fwrite(fid,uniprotDL);
-    fclose(fid);
-    clear uniprotDL % Load file fresh
+filename = [num2str(taxonID) '.tsv'];
+if ~exist(fullfile(geckoPath,'databases/uniprot/',filename))
+    disp(['Downloading Uniprot data for taxonomic ID ' taxonID '. This can take a few minutes.'])
+    %url = ['https://www.uniprot.org/uniprot/?query=taxonomy:' taxonID ...
+    %    '&format=tab&compress=no&columns=id,genes(OLN),genes(PREFERRED),ec,database(GeneID),database(RefSeq),mass,sequence'];
+    url = ['https://rest.uniprot.org/uniprotkb/stream?query=taxonomy_id:' num2str(taxonID) ...
+           '&fields=accession%2C' param.uniprotGeneIdField ...
+           '%2Cgene_primary%2Cec%2Cxref_geneid%2Cxref_refseq%2Cmass%2Csequence&format=tsv&compressed=false&sort=protein_name%20asc'];
+    urlwrite(url,fullfile(geckoPath,'databases/uniprot/',filename));
+%    options=weboptions('Timeout',100); %If the code fails, this may need to be increased
+%    uniprotDL = webread(url,options);
+%    dl2 = decode(uniprotDL);
+%    fid=fopen(fullfile(geckoPath,'databases/uniprot/',filename),'w');
+%    fwrite(fid,uniprotDL);
+%    fclose(fid);
+%    gunzip(fullfile(geckoPath,'databases/uniprot/',filename))
+%    clear uniprotDL % Load file fresh
 else
-    disp(['Load existing Uniprot database file for taxonomic ID ' taxonID '.'])
+    disp(['Loading existing Uniprot database file for taxonomic ID ' taxonID '.'])
 end
 
-fid         = fopen(fullfile(geckoPath,'databases','uniprotDB.tsv'),'r');
+fid         = fopen(fullfile(geckoPath,'databases/uniprot/',filename),'r');
```

Comment:
Not for this PR, but ideally any model-specific files should not be stored in the GECKO folder structure. The ModelAdapter could include a parameter that specifies the path where model-specific files should be stored? This also affect other functions (for instance `getComplexData`), and is probably best addressed in a separate PR.

Reply:
I agree that model-specific files should be stored in an "external" folder. I suggest this structure. 

<img width="1104" alt="Screenshot 2022-11-30 at 10 01 13" src="https://user-images.githubusercontent.com/39610893/208209553-727987c1-0324-431d-bf13-d28d7a9b0d95.png">

the idea is that projects folder (can be also call Custom ? ) contains the data/scripts for model-specific files it a path is not defined in getModelParameters.m. 

getModelParameters.m also includes a path where to store the results ecModel/prot_model, this is usefull when working with git repos locally. 

I already implemented this approach by using GECKO 2 (in my fork). Maybe it can be implemented for GECKO 3 ?

Reply:
So, in my mind, it is ok to have it there, but it could also be somewhere else as long as it is a place that we can guarantee will be available. Keep in mind that these files are not part of the repo - they are downloaded on the fly - it is only the folder that is available. The code checks for if the file with the same name as the taxon id is in the folder, cached there from previous runs. If not, it downloads and saves the file in the folder. So, this is in my mind not model-specific in that sense, it is only the cached files that are downloaded that are specific.

Reply:
But I mean, not only one specific file, GECKO also have many model/oganism specific functions which I think should be in other folder. I mean there should be a src folder with the GECKo functions that no need change by the user

Reply:
Let's have this discussion in a separate Issues, will open one now.

Reply:
Now in #187. This discussion is a bigger issue, but should not keep this PR from merging.